### DISCUSSION TITLE
Automatic flashcard generation

### DISCUSSION THREAD

**Discussion by vasnt** [AlexErrant/Pentive#4] [Ideas]
would love if AI feature like Automatic flashcard generation, labelling cards can be added.

**Comment by AlexErrant** [comment]
Hm, I can't tell if you're asking for AI features in general, or if you just want AI-generated flash cards. If you're looking for AI features in general, I don't really think that way. I like to think of problems first and _then_ their possible solutions.

For automatic card generation, I'm fairly bearish on AI-generated cards. Good cards are hard to write, and I'm not convinced AI would be able to capture the nuance of a good card. Gathering the training data and then training the models isn't something that I'm particularly interested in, at least in the foreseeable future. I would have access to the output data (the Question/Answer prompts), but not the input data (the raw (potentially copyrighted) textbook/reference material). Still, this is why there are plugins - this would be fairly trivial to write a plugin _once the model exists_ - you can do whatever you want client-side! I like what StackOverflow does with [releasing datasets](https://stackoverflow.blog/2014/01/23/stack-exchange-cc-data-now-hosted-by-the-internet-archive/) - I plan on making the data available, so if someone else wants to train a model they'll know what I know. I'm open to changing my mind - please feel free to persuade me! Most of this thought process applies to any client-side AI, e.g. auto-tagging/classification, text-to-speech, speech-to-text, etc. Plugins could give all of these capabilities; AI is just an impl detial. More generally, AI generated cards seems to be trying to solve the problem of "writing cards is difficult", which I totally get. However, I feel like "people sharing cards they've written" is a better solution than "use generative AI".

I _do_ see a place for AI on the server, for example a recommendation algorithm similar to YouTube's. Something I'm intrigued by is the possibility of automatically detecting poorly written cards. If, say, 25% of students keep hitting "Bury/Hard/Again" on a card, that could be a sign to the author that the card needs rewording... but that would require people to upload their reviews to my server, which is an area of privacy concern. I haven't thought about this too much - I still need to build the basic functionality 😅 This project in its current state is very much unusable - it doesn't even have a way to review cards!

**Comment by vasnt** [comment]
- i understand sharing cards is efficient solution, but there are many stuffs users studies themselves, & may be difficult to find required specific flashcards from web. So it is always better to have some bad/good (editable- AI generated) flashcards rather than no cards at all. So later on, if needed, user can edit it(words/phrases etc..) 
- Of course, crucial flashcards needs to be generated manually, but obvious/simple (AI generated) cards can be useful for beginner levels.

**Comment by AlexErrant** [comment]
What kind of interface/input are you envisioning? Something like ChatGPT, i.e. "generate basic flashcards for Spanish and English words"? Would an external resource be involved - i.e. "reference this blog post on common Spanish/English wordpairs"?

**Comment by vasnt** [comment]
i thought of basic flashcards as you mentioned.  
Other additions can be

-  as you thought, referencing other blog post/external sources/docs
- cards based on given example in prompt

**Comment by AlexErrant** [comment]
This guy is doing something like what you want https://www.reddit.com/r/Anki/comments/16itqcb/i_built_an_experimental_spaced_repetition_app_to/

Also

* https://www.reddit.com/r/Anki/comments/16km11j/any_ai_software_to_make_cards_from_highlighting_a/
* https://www.ace-it.app/

**Comment by vasnt** [comment]
Thanks for link. Keep sharing.
